nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -74,6 +74,27 @@ WaypointFollower::on_configure(const rclcpp_lifecycle::State & /*state*/)
     get_node_waitables_interface(),
     "FollowWaypoints", std::bind(&WaypointFollower::followWaypoints, this));
 
+  // related to GPS waypoint following
+  waypoint_follower_client_node_ = std::make_shared<rclcpp::Node>(
+    "_", "", rclcpp::NodeOptions().arguments(new_args));
+
+  waypoint_follower_action_client_ = rclcpp_action::create_client<ClientTGPS>(
```

Comment:
Ah, I see why you think you need the extra node, you're calling the action within the action. This is what I was mentioning on refactoring the waypoint follower action callback itself (https://github.com/ros-planning/navigation2/blob/c6ee43704ee67f41872dfafe527f69fa8eca4d3c/nav2_waypoint_follower/src/waypoint_follower.cpp#L166) and getting the action server inputs and then calling a set of breakout functions to do the following work.

Then the GPS waypoint follower can do the exact same thing; get the inputs, convert to map form, then call the exact same internal functions for the exact same behaviors! That way, the GPS WP doesn't have to call the cartesian WP action server because they share the same internal logic. 

Extra nodes on the network add really really non-trivial overhead, so we need to minimize the use of them when possible. Plus having 2 servers in a class sharing some functions is much clearer architecturally than a server's action  calling another server-owned action.